src/jspdf.js

```diff
@@ -3878,14 +3878,19 @@ function jsPDF(options) {
     var doReversing = typeof options.R2L === "boolean" ? options.R2L : R2L;
     if (doReversing === true) {
       text = processTextByFunction(text, function(text, posX, posY) {
-        return [
-          text
-            .split("")
-            .reverse()
-            .join(""),
-          posX,
-          posY
-        ];
+        const rtlLangCharsRegex = /[\u0590-\u05FF\u0621-\u064A]/;
+        if (rtlLangCharsRegex.test(text)) {
+          return [
+            text
+              .split("")
+              .reverse()
+              .join(""),
+            posX,
+            posY
+          ];
+        } else {
+          return [text, posX, posY];
+        }
```

Comment:
While this change somehow seems to fix the issue, I suggest a different one: If I'm correct, the bidiEngine is just not invoked with the correct parameters. When I set the bidi options to 
```
options.isInputVisual = false;
options.isOutputVisual = true;
options.isInputRtl = false;
options.isOutputRtl = false;
```
everything seems to work correctly, if I set
```
doc.setR2L(false);
```
If I understand correctly, the jsPDF R2L option does not specify the base direction of the text. Instead, it specifies if the logical order of the characters in the string is RTL.

Reply:
Is this fix suggestion works for u with hebrew/rtl language fonts?

Reply:
This works for me:
```
const doc = new jsPDF();
doc.addFont(
  "https://cdn.jsdelivr.net/npm/rubik-font@0.0.3/fonts/Rubik-Regular.ttf",
  "Rubik",
  "normal"
);

doc.setFont("Rubik"); // set font
doc.setFontSize(50);
doc.setR2L(false);

const mixedRtlLtrText = "שלום עליכם 123";

const options = {
  isInputVisual: false,
  isOutputVisual: true,
  isInputRtl: false,
  isOutputRtl: false
};

doc.text(mixedRtlLtrText, 10, 20, options);
doc.text("1234 ", 10, 40, options);
doc.text("1234", 100, 40, options);
doc.text(" שלום עליכם ", 10, 60, options);
doc.text("abc ", 10, 80, options);
doc.text("abc", 100, 80, options);

doc.save();
```

![image](https://user-images.githubusercontent.com/16103654/132648988-727d3f20-9f28-4f43-8709-c7ebeeab9ea6.png)

Is the outcome correct? I'm not a Hebrew/RTL expert...

Reply:
Yes, it is correct.
Do you know how to make it work with autotable?

Reply:
@liorokach Not sure. We probably need to fix it in the jsPDF library and then it will work without passing custom options to the bidi engine.

@ronbiter Could you make the suggested changes?

Reply:
Hey, any updates on this? I installed the forked ronnibitter repo and mixed content shows up correctly, but with autotable it's still mixed up.
Let me know if I can help

Reply:
hey i'm sorry I don't have the time currently to go back to this. the bidi engine did work when I tested it a while ago but it wasn't perfect at all situations so I kept with my change with few minor tweaks for different situations that we encountered.
you are all welcome to fork the repo and PR it to the project. I don't know currently when I will have the time fix the PR (it is still failing the tests).
*** this fix was mainly for generating PDF from an html code (using html2canvas) didn't test it for other methods.

Reply:
@HackbrettXXX worked for me!